### PR TITLE
Better error handling

### PR DESCRIPTION
**Pull request by pascutto**
This refactors the error handling to remove all the exceptions (and exception printers registrations), which were not suitable for a library usage of Gospel.
All errors now raise an `Error of loc * kind`, which is easy to catch and print by client libraries

### CODE REVIEW

File: src/Opprintast.ml [hunk 1]
```diff
@@ -11,7 +11,7 @@
 (*                                                                        *)
 (*   All rights reserved.  This file is distributed under the terms of    *)
 (*   the GNU Lesser General Public License version 2.1, with the          *)
-(*   special exception on linking described in the file LICENSE.          *)
+(*   special exception on linking described in the file LICENSW.          *)
```
Comment (Armael) [inline src/Opprintast.ml:14]: erroneous change?

File: src/coercion.ml [hunk 2]
```diff
@@ -5,14 +5,13 @@
 (*                                                                  *)
 (*  This software is distributed under the terms of the GNU Lesser  *)
 (*  General Public License version 2.1, with the special exception  *)
-(*  on linking described in file LICENSE.                           *)
+(*  on linking described in file LICENSW.                           *)
```
Comment (Armael) [inline src/coercion.ml:8]: same

File: src/parser_frontend.mli [hunk 3]
```diff
@@ -5,37 +5,34 @@
 (*                                                                  *)
 (*  This software is distributed under the terms of the GNU Lesser  *)
 (*  General Public License version 2.1, with the special exception  *)
-(*  on linking described in file LICENSE.                           *)
+(*  on linking described in file LICENSW.                           *)
```
Comment (Armael) [inline src/parser_frontend.mli:8]: same

**Review by Armael — COMMENTED** [review]
(small refactoring bug)

File: src/Opprintast.ml (hunk 1 above)
Comment (pascutto) [reply src/Opprintast.ml]: Ah yes, my `sed` was a bit too brutal. Thanks for catching this!

**Pull request merged**